lib/fencing/st_client.c

```diff
@@ -2511,23 +2511,38 @@ stonith__event_exit_reason(stonith_event_t *event)
  * \return Newly allocated string with description of \p event
  * \note The caller is responsible for freeing the return value.
  *       This function asserts on memory errors and never returns NULL.
- * \note This currently is useful only for events of type
- *       T_STONITH_NOTIFY_FENCE.
  */
 char *
 stonith__event_description(stonith_event_t *event)
 {
     const char *origin = event->client_origin;
+    const char *origin_node = event->origin;
     const char *executioner = event->executioner;
+    const char *device = event->device;
+    const char *action = event->action;
+    const char *target = event->target;
     const char *reason = stonith__event_exit_reason(event);
     const char *status;
 
+    // Use somewhat readable defaults
     if (origin == NULL) {
         origin = "a client";
     }
+    if (origin_node == NULL) {
+        origin_node = "a node";
+    }
     if (executioner == NULL) {
         executioner = "the cluster";
     }
+    if (device == NULL) {
+        device = "unknown";
+    }
+    if (action == NULL) {
+        action = (event->operation == NULL)? "(unknown)" : event->operation;
+    }
+    if (target == NULL) {
+        target = "no node";
+    }
```

Comment:
now that it is getting longer using ternary operator in the first assignment might lead to a short readable translation table (action being the exception unfortunately).

Reply:
Most of the lines would run over 80 characters with a ternary

I wonder if it might be worthwhile to have a macro like crm_str() but taking a default, like

    const char *origin = pcmk__s(event->client_origin, "a client");

Reply:
will do separate pr for str macro